airflow/hooks/dbapi.py

```diff
@@ -211,8 +211,13 @@ def _run_command(self, cur, sql_statement, parameters):
         self.log.info("Running statement: %s, parameters: %s", sql_statement, parameters)
         if parameters:
             cur.execute(sql_statement, parameters)
+            res = cur.fetchall()
+            result={'sql_statement': res,'headers': list(map(lambda t: t[0], cur.description))}
+            self.log.info("Query Results: %s", result)
```

Comment:
0This will actually impact ALL DB operators - not only Snowflake. I think info level here is quite a bit too much - possibly Debug will be better  - also running fetchall is potentially impacting the performance and seems we only need it to print logging information.

I'd change it to debug level and put `fetchall` under:

```
if self.log.isEnabledFor(logging.DEBUG):
```

to minimise side-effects

Reply:
Hi @potiuk  !! 

something like that? 

if parameters:
            cur.execute(sql_statement, parameters)

            if self.log.isEnabledFor(logging.DEBUG):
     
                res = cur.fetchall()
                result={'sql_statement': res,'headers': list(map(lambda t: t[0], cur.description))}
                self.log.info("Query Results: %s", result)

Reply:
What kind of query do you run that this change is needed?
In ETLs most queries that you run are DMLs (insert, update, delete, etc..) these statments return 1 row.
I think you are running DQL (select) which is a bit odd?
My question here is if this change is needed and if so why not using fetchone?

Reply:
Hi @eladkal 

You're right respect that ETLs most queries that you run are DML (insert, update, delete, etc..) but when we need to transform the query results in order to send an S3.

Reply:
Just a comment to what @eladkal wrote. It's not entirely correc. I think the DML mostly is pretty valid for the operators. But we do not actually prevent people to run the DQL queries, via Hooks. It's no problem whatsoever to build your custom operator to use two hooks and pass the DQL'ed data with some manipulation to another hook. This is what most transfer operators do. 

That's why I think having debug level to show the results is pretty useful when you iterate and test your custom operators.

Reply:
Thanks for your feedback

Reply:
I'm ok with the debug. my issue is mostly with the `fetchall`
It's important to remember that debug isn't for a specific dag - it's cluster wide setting so when you set debug other DAGs log may get really big but I guess that depends how each organization works.

Reply:
AFAIK - fetchall has no effect on DQK/DDL queries :). But I agree for DQL queries it can have disastrous consequences by fetching all data to memory. But IMHO this is not a a problem - it's expected that when you debug, you can run out of memory or things will start running slower etc. I do not think fetchall changes the actual semantics of what happens with the results, so memory/time needed to fetch the records should be the only side-effect (but also that's why it should only run when debug level is effective for the logger:

```
if self.log.isEnabledFor(logging.DEBUG):
```

Reply:
> Hi @eladkal
> 
> You're right respect that ETLs most queries that you run are DML (insert, update, delete, etc..) but when we need to transform the query results in order to send an S3.

I don't understand your use case. Could you explain a little bit, please? Just curious